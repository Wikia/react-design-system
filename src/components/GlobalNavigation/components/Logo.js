import React from 'react';
import PropTypes from 'prop-types';

const Logo = ({href, edition}) => (
  <a
    href={href}
    className="wds-global-navigation__logo"
    data-tracking-label="logo"
  >
    <svg className="wds-global-navigation__logo-image wds-is-wds-company-logo-fandom-white">
      <use xlinkHref="#wds-company-logo-fandom-white" />
    </svg>
    {edition && (
      <span className="wds-global-navigation__edition-text">{edition}</span>
    )}
  </a>
);

Logo.propTypes = {
  edition: PropTypes.string.isRequired,
  href: PropTypes.string.isRequired,
};

Logo.defaultProps = {
  edition: null,
};

export default Logo;
