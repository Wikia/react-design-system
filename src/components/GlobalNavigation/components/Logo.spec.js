import React from 'react';
import renderer from 'react-test-renderer';

import Logo from './Logo';

test('Logo should render logo without edition', () => {
  const component = renderer.create(
    <Logo href="http://example.com" />
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test('Logo should render logo with edition', () => {
  const component = renderer.create(
    <Logo href="http://example.com" edition="UK" />
  );
  expect(component.toJSON()).toMatchSnapshot();
});
