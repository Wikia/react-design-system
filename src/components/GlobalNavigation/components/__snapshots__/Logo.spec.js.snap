// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo should render logo with edition 1`] = `
<a
  className="wds-global-navigation__logo"
  data-tracking-label={undefined}
  href="http://example.com"
  title="FANDOM"
>
  <svg
    className="wds-global-navigation__logo-image wds-is-wds-company-logo-fandom-white"
  >
    <use
      xlinkHref="#wds-company-logo-fandom-white"
    />
  </svg>
  <span
    className="wds-global-navigation__edition-text"
  >
    UK
  </span>
</a>
`;

exports[`Logo should render logo without edition 1`] = `
<a
  className="wds-global-navigation__logo"
  data-tracking-label={undefined}
  href="http://example.com"
  title="FANDOM"
>
  <svg
    className="wds-global-navigation__logo-image wds-is-wds-company-logo-fandom-white"
  >
    <use
      xlinkHref="#wds-company-logo-fandom-white"
    />
  </svg>
</a>
`;
